reVX/x_mission_paths/file_handlers.py

```diff
@@ -0,0 +1,271 @@
+
+        self._plot_costs_arr = None
+
+        logger.debug(f'Loading SC points from {sc_points_f}')
+        # TODO - make this resolution aware
+        self.sc_points = self._load_sc_points(sc_points_f)
+
+    def _load_sc_points(self, sc_points_f, raw_crs='epsg:4326'):
+        """
+        Load supply curve points from disk
+
+        Parameters
+        ----------
+        sc_points_f : String
+            Path to supply curve points CSV
+        raw_crs : String
+            CRS string for SC points file
+
+        Returns
+        -------
+        sc_points : List of SupplyCurvePoint
+        """
+        pts = pd.read_csv(sc_points_f)
+        geo = [Point(xy) for xy in zip(pts.longitude, pts.latitude)]
+        pts = gpd.GeoDataFrame(pts, crs=raw_crs, geometry=geo).to_crs(self.crs)
+
+        sc_points = []
+        for _, row in pts.iterrows():
+            sc_pt = SupplyCurvePoint(row.sc_point_gid, row.sc_row_ind,
+                                     row.sc_col_ind, row.geometry, self.rct,
+                                     self.regions_arr)
+            sc_points.append(sc_pt)
+        return sc_points
+
+    @property
+    def plot_costs_arr(self):
+        """
+        Return costs array for plotting with transmission barriers set to -1
+        """
+        if self._plot_costs_arr is None:
+            self._plot_costs_arr = self.costs_arr.copy()
+            self._plot_costs_arr[self._barriers_arr == BARRIERS_MULT] = -1
+
+        return self._plot_costs_arr
+
+
+class AllConnsLoader:
+    def __init__(self, all_conns_f, subs_f='_substations_cache.shp'):
+        """
+        Load substations, transmission lines, load centers, and sinks from disc
+
+        Parameters
+        ----------
+        all_conns_f : String
+            Path to conus_allcons.gpkg
+        subs_f : String
+            Path to substations shapefile that is autogenerated by this class
+        """
+        conns = gpd.read_file(all_conns_f)
+        conns = conns.drop(['bgid', 'egid', 'cap_left'], axis=1)
+
+        self.t_lines = conns[conns.category == 'TransLine']
+        self.lcs = conns[conns.category == 'LoadCen']
+        self.sinks = conns[conns.category == 'PCALoadCen']
+
+        subs_f = os.path.join(os.path.dirname(all_conns_f), subs_f)
+        if os.path.exists(subs_f):
+            logger.debug(f'Loading cached substations from {subs_f}')
+            self.subs = gpd.read_file(subs_f)
+        else:
+            self.subs = conns[conns.category == 'Substation']
+            self._update_sub_volts()
+            self.subs.to_file(subs_f)
+
+    def _update_sub_volts(self):
+        """
+        Get substation voltages from trans lines
+        """
+        logger.debug('Determining voltages for substations, this will take '
+                     'a while')
+        self.subs['temp_volts'] = self.subs.apply(self._get_volts, axis=1)
+        volts = self.subs.temp_volts.str.split('/', expand=True)
```

Comment:
I try to avoid the parameter notation in pandas/geopandas as it is confusing. Is temp_volts a method? Oh wait its a column. Lets just use column notation.